pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1216,15 +1217,17 @@ public void resetCursorOnPosition(
             @ApiParam(value = "Is authentication required to perform this operation")
             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative,
             @ApiParam(name = "messageId", value = "messageId to reset back to (ledgerId:entryId)")
-            MessageIdImpl messageId) {
+                    ResetCursorData resetCursorData) {
```

Comment:
How did you maintain backward compatibility on this endpoint?

Reply:
> How did you maintain backward compatibility on this endpoint?

Now the properties of this object are consistent with the previous `MessageId`, and there is no problem with Json deserialization.

Reply:
Can you add an integration test for that?

Reply:
> Can you add an integration test for that?

OK